uwblueprint/ccbc
# Create New Page For Editing Reviews

## Notion ticket link

[Edit Review: Load Data](https://www.notion.so/uwblueprintexecs/Edit-Review-Load-Data-4454f563305046e0940842a8b3b08627)



## Implementation description
* Converted the pre-existing `CreateReview` page into a component that takes in `id` as an optional prop
* Created a new `CreateReviewPage` that contains the `CreateReview` component without an id passed in
* Created a new `EditReviewPage` that contains the `CreateReview` component with an id passed in
* Refactored and added functionality to `CreateReview` to support editing



## Steps to test
1. Go to `/create-review` and ensure it works the same as before
2. Create a new review using the API - sample request body below
3. Go to `/edit-review/:id` where `id` is from the response of Step 2 and test that the fields get pre-populated from the review you created



```
{
    "body": "test",
    "byline": "test",
    "featured": true,
    "createdBy": 3,
    "books": [
        {
            "title": "Shadow Chasers",
            "coverImage": "https://images-na.ssl-images-amazon.com/images/I/51WCelNAo1L.jpg",
            "titlePrefix": "The",
            "seriesOrder": "1",
            "illustrator": [
                "Spencer Fencer"
            ],
            "translator": [
                "Translator Alligator"
            ],
            "formats": [
                {
                    "format": "Soft cover",
                    "isbn": "42069",
                    "price": "13"
                }
            ],
            "minAge": 4,
            "maxAge": 8,
            "authors": [
                {
                    "fullName": "Sophia Spencer",
                    "displayName": "Sophia Spencer",
                    "attribution": "Author"
                }
            ],
            "publishers": [
                {
                    "fullName": "Publisher Fisher",
                    "publishYear": 2012
                }
            ],
            "seriesName": "The Amazing Series"
        }
    ],
    "tags": [],
    "publishedAt": 1636226732806
}
```



## What should reviewers focus on?
* 


## Checklist
